### PR TITLE
Api provides user to complete a shopping trip

### PR DESCRIPTION
**Pull request by emtalen**
### User Story
```
As a resident 
In order to complete a trip
I would like to be able to interact with accepted requesters
```
## Changes proposed in this pull request:
* Modifying update method to set ping to completed
* Adding show method for pong for user to see their request
* Modifying update method for pong to update with the total cost
* Fixing pong id in destroy action
* Restricting so user only can have one un completed ping at the time

## What I have learned working on this feature:
*  Small changes can make a big difference
* Practising refactoring

### CODE REVIEW

File: app/controllers/pings_controller.rb [hunk 1]
```diff
@@ -53,14 +56,37 @@ def ping_params
...
+  end
+
+  def correct_user?
+    @requested_ping = Ping.find(params[:id])
+    if @requested_ping.user_id == current_user.id
+    else
+      render json: {
+        message: "You are not authorized to update another user's ping"
+      },
+             status: 401
```
Comment (kfrostare) [inline app/controllers/pings_controller.rb:75]: is status supposed to be double indented? if yes, ignore me.

Comment (emtalen) [reply app/controllers/pings_controller.rb:75]: Its the formatter that likes it that way 🤪

**Review by johanbounce — APPROVED** [review]
Every single line of code was so good!!!!

**Review by kaylawoodbury — APPROVED** [review]
This looks good to me, great  work guy! Keep up the good work!

**Pull request merged**